http/http_client.hpp

```diff
@@ -363,37 +387,56 @@ class HttpClient : public std::enable_shared_from_this<HttpClient>
             BMCWEB_LOG_ERROR
                 << "Maximum number of retries reached for Subscriber:" << subId;
 
-            // Clear queue.
-            while (!requestDataQueue.empty())
```

Comment:
Why duplicate this in 3 places now?

Reply:
We are clearing entire queue in case of Terminate/Suspend. For RetryForever, we are not cleaning the complete queue with a while loop.